waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/AbstractMetaStore.java

```diff
@@ -161,6 +164,23 @@ public void setMappedDatabases(List<String> mappedDatabases) {
     this.mappedDatabases = mappedDatabases;
   }
 
+  public Map<String, String> getDatabaseNameMapping() {
+    return databaseNameMapping;
+  }
+
+  public void setDatabaseNameMapping(Map<String, String> databaseNameMapping) {
+    if (databaseNameMapping == null) {
```

Comment:
Would it be better to initialise the databaseNameMapping at the class property level and avoid the initialisation in this setter?

Reply:
if I do it at the class level then it can get overwritten by the setter I just never want this to be null

Reply:
Right, I see.

I don't know if calling the getter before the setter is a valid use case. The class would be in an illegal state and you'd get a `null` rather than an empty map.

I think I'd probably change the setter to throw an exception if a `null` value is provided, and instantiate the collection at the property level. It gives a much better API in my opinion.

It's fine if you leave it like this too :)

Reply:
How about I change the getter to return empty if null. then we cover that case as well